src/main/java/careeratlas/backend/Service/JobSearchServiceImpl.java

```diff
@@ -1,34 +1,36 @@
 package careeratlas.backend.Service;
 
-import careeratlas.backend.Domain.JobResponse;
+import careeratlas.backend.Domain.IndeedResponse;
 import careeratlas.backend.Domain.JobSearch;
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.databind.ObjectMapper;
 import org.springframework.stereotype.Service;
-
-import java.util.ArrayList;
+import java.io.IOException;
 import javax.ws.rs.client.*;
 import javax.ws.rs.core.MediaType;
 import javax.ws.rs.core.Response;
-import java.util.HashMap;
 
 @Service
 public class JobSearchServiceImpl implements JobSearchService {
 
+    private Client client = ClientBuilder.newClient();
```

Comment:
Were you thinking something like this?

Reply:
Yeah, actually crazy hard to find the documentation for it but I'm fairly certain the creation of a client is fairly heavy. you want to share them as much as possible. So if you need them in other classes consider creating a bean to inject everywhere. If its just that one class, make that field static and final and youre set.

Reply:
https://stackoverflow.com/questions/34566278/how-to-correctly-share-jax-rs-2-0-client